packages/core/src/extension/resizable-iframe-creator.ts

```diff
@@ -90,5 +93,12 @@ export default class ResizableIframeCreator {
 
             window.addEventListener('message', handleMessage);
         });
+
+        return detachmentObserver.ensurePresence([iframe], promise);
```

Comment:
I have tried to call `failedCallback` method in this way, but it was not ideal.
```
return detachmentObserver.ensurePresence([iframe], promise).catch((error) => {
            failedCallback();

            throw error;
        });
```
Because in our extension implementation, `failedCallback` assumes that the iframe is still present.
https://github.com/bigcommerce/checkout-sdk-js/blob/2e6f95f3718a37ef48e1c479136a699e95e6db6f/packages/core/src/extension/extension-iframe.ts#L44

Reply:
I see you are throwing an exception here. What happens on checkout side? Does the exception get caught?

Reply:
On checkout-js side, we didn't catch it yet.

https://github.com/bigcommerce/checkout-js/blob/6fd4945e82c3d5748014e4d55d2a382a5837858d/packages/checkout-extension/src/ExtensionService.ts#L61

Is it a good idea that we catch it and ignore the error if it is `UnexpectedDetachmentError`?

Reply:
I suppose we should still handle the exception, maybe a logger message. Atleast so we don't blow up checkout.

Reply:
Preferably unhandled promise rejections should be caught, as they are typically flagged by error tracking tools such as Sentry and Noibu, even if they don’t impact user experience.

Reply:
Hi @animesh1987 and @davidchin, I have added a try/catch to handle errors in `failedCallback`.

On `checkout-js` side, I opened a PR to catch errors in `checkoutService.renderExtension`. This PR does not depend on SDK. https://github.com/bigcommerce/checkout-js/pull/2105

Reply:
Updated testing video.

https://github.com/user-attachments/assets/2022684f-608f-47f7-9337-3667ecca2180